src/framing/FramedReader.cpp

```diff
@@ -66,35 +69,31 @@ size_t FramedReader::readFrameLength() const {
   return frameLength;
 }
 
-void FramedReader::onSubscribeImpl(
-    std::shared_ptr<Subscription> subscription) noexcept {
-  CHECK(!streamSubscription_);
-  streamSubscription_ = std::move(subscription);
-  frames_->onSubscribe(shared_from_this());
+void FramedReader::onSubscribe(
+    yarpl::Reference<Subscription> subscription) noexcept {
+  SubscriberBase::onSubscribe(subscription);
+  subscription->request(kMaxRequestN);
```

Comment:
I have synced with Ondrej. We will keep this line as is and update Http2StreamHandler.